target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

```diff
@@ -40,6 +40,19 @@ askey_do_upgrade() {
 	nand_do_upgrade "$1"
 }
 
+vboot_do_upgrade() {
+	local tar_file="$1"
+
+	local board_dir=$(tar tf $tar_file | grep -m 1 '^sysupgrade-.*/$')
+	board_dir=${board_dir%/}
+
+	echo "Preparing to flash \"${board_dir}\" to /dev/mmcblk0p{1,2}"
+	ask_bool 0 "Abort" && exit 1
+
+	tar Oxf "${tar_file}" "${board_dir}/kernel" | dd of=/dev/mmcblk0p2 bs=1M
```

Comment:
could this be a candidate for emmc_upgrade_tar()?

This is new, you can check it out @ https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/upgrade/emmc.sh

Reply:
Perhaps! Frankly I didn't actually spend much time on sysupgrade so it might not be working at all. I may give it another go and try out those helpers, so thanks for the pointer.

Reply:
Awesome, this works.